### PR TITLE
Adding Py Dependency

### PR DESCRIPTION
**Pull request by psteyer**
Adding Py dependency see following Jira ticket for more information:
https://anaconda.atlassian.net/browse/CVE-566

## changes
- adding `py`
- bumping build #
- updated to the newer version
- removed no arch

### CODE REVIEW

**Review by ryanskeith — APPROVED** [review]
LGTM.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -22,6 +22,7 @@ requirements:
   run:
     - python
     - pytest >=3.1.0
+    - py
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:25]: Could you please add a comment that CVE-2022-42969 does not apply to pytest-fork, as it does not use `py.svn`, see https://github.com/pytest-dev/pytest-forked/issues/80#issuecomment-1381755809?

**Review by skupr-anaconda — APPROVED** [review]
Could you also update it to the latest version 1.6.0?

**Comment by psteyer** [comment]
I updated to the newer version, and removed no arch

**Review by skupr-anaconda — APPROVED** [review]

**Review by ryanskeith — APPROVED** [review]
LGTM

**Pull request merged**